test/OpenTelemetry.Exporter.Zipkin.Tests/ZipkinActivityExporterTests.cs

```diff
@@ -127,6 +129,30 @@ public async Task ZipkinActivityExporterIntegrationTest(bool useShortTraceIds)
                 Responses[requestId]);
         }
 
+        [Fact]
+        public void UseZipkinActivityExporterWithCustomActivityProcessor()
+        {
+            const string ActivitySourceName = "zipkin.test";
+            Guid requestId = Guid.NewGuid();
+
+            var openTelemetrySdk = OpenTelemetrySdk.EnableOpenTelemetry(b => b
+                            .AddActivitySource(ActivitySourceName)
+                            .UseZipkinActivityExporter(
+                                o =>
+                            {
+                                o.ServiceName = "test-zipkin";
+                                o.Endpoint = new Uri($"http://{this.testServerHost}:{this.testServerPort}/api/v2/spans?requestId={requestId}");
+                            }, p => p.AddProcessor((next) => new TestProcessor(next))));
+
+            var source = new ActivitySource(ActivitySourceName);
+            var activity = source.StartActivity("Test Zipkin Activity");
+            activity?.Stop();
+
+            Assert.Equal("zipkinCustomTag", activity?.Tags.FirstOrDefault().Key);
```

Comment:
Can you do something like this?
https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/test/OpenTelemetry.Tests/Implementation/Trace/ActivitySourceAdapterTest.cs#L56

So that this test logic controls what the Processor is doing.

Reply:
To confirm that the Testprocessor is added to the same pipeline who end is ZIpkin, I'd add a test where the Testprocessor doesn't call its Next. We'd then validate Zipkin got nothing in that case.

Reply:
Added processor logic within the test.